### DISCUSSION TITLE
viewport / unproject

### DISCUSSION THREAD

**Discussion by TextTL** [visgl/deck.gl#7179] [General]
Hi, 

I'm having an issue with unprojecting screen coordinate to the world coordinate after a zoom. I'm trying to change a path coordinate according to the position of the mouse pointer. When unprojecting, I used pickInfo.viewport.unproject (from the picked object); things worked fine until the zoom level was changed with the mouse wheel. pickInfo.viewport.unproject did not work correctly any longer. This is understandable. 

Do you know how can I get the viewport instance with the most up-to-date zoom level so that I can unproject the coordinate correctly every time?  I'm using React DeckGL component. 

Thank you!

**Comment by TextTL** [comment]
After looking into deck object from ref, I found that I could access the viewport from deck.layerManager.context.viewport.unproject.

Thank you all,

**Reply by Pessimistress** [reply]
If you are using TypeScript, the public API is `deck.getViewports()`.
